### PR TITLE
fix#279700 Corruption and bad links using exchange voices on score with parts

### PR DESCRIPTION
**Pull request by Rockettwo**
See [279700](https://musescore.org/en/node/279700). 
The exchange voices part probably needs a review, but it worked for the original problem described in the issue and Marcs simple example.

### CODE REVIEW

**Comment by anatoly-os** [comment]
@Rockettwo could you please squash the commits? Also, add whitespace after "fix" to automatically close the issue.

File: libmscore/edit.cpp [hunk 1]
```diff
@@ -3727,30 +3727,39 @@ void Score::undoExchangeVoice(Measure* measure, int v1, int v2, int staff1, int
                   int stTrack = st->idx() * VOICES;
                   Measure* m = st->score()->tick2measure(tick);
                   Excerpt* ex = st->score()->excerpt();
+
                   if (ex) {
                         QMultiMap<int, int> t = ex->tracks();
                         QList<int> ts = t.values(s);
                         QList<int> td = t.values(d);
 
+                        // ts... source tracks, td...destination tracks
```
Comment (anatoly-os) [inline libmscore/edit.cpp]: Just rename the variables! Rename ts to sourceTracks and others :)

File: libmscore/edit.cpp [hunk 2]
```diff
@@ -3727,30 +3727,39 @@ void Score::undoExchangeVoice(Measure* measure, int v1, int v2, int staff1, int
...
+                        // tss... source track, tdd...destination track
                         for (int tss : ts) {
+                              // don't care about other linked staves
                               if (!(stTrack <= tss) || !(tss < stTrack + VOICES))
                                     continue;
 
                               int temp = t.key(tss);
                               QList<int> test = t.values(temp + diff);
-                              bool hasVoice = false;
+                              bool hasVoice = false; //
```
Comment (anatoly-os) [inline libmscore/edit.cpp]: Debugging artefact? `//`

**Comment by anatoly-os** [comment]
LGTM, so once you rename the variables and remove empty comment, I'll merge it :)

**Comment by Rockettwo** [comment]
Was a joy to update the variable names :)

**Pull request merged**